### PR TITLE
Delay starting the work to scan for todo-items

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Potential fix for https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems/edit/1466973.

### CODE REVIEW

File: src/VisualStudio/Core/Def/TodoComments/VisualStudioTodoCommentsService.cs [hunk 1]
```diff
@@ -77,15 +77,20 @@ public void Dispose()
...
 
-        private async Task StartAsync()
+        private async Task StartAsync(Workspace workspace)
         {
             // Have to catch all exceptions coming through here as this is called from a
             // fire-and-forget method and we want to make sure nothing leaks out.
             try
             {
+                // Don't bother doing anything until the workspae has actually loaded.  We don't want to add to any
+                // startup costs by doing work too early.
```
Comment (dibarbet) [inline src/VisualStudio/Core/Def/TodoComments/VisualStudioTodoCommentsService.cs:90]: should we consider not doing anything at all until someone opens the todo comment window?   I don't think I've ever opened it, so for people like me it seems like there is no point in this even running.

**Review by dibarbet — APPROVED** [review]
makes sense to me.

File: src/Features/LanguageServer/Protocol/Features/TodoComments/TodoCommentsListener.cs [hunk 2]
```diff
@@ -64,14 +70,10 @@ public void Dispose()
...
+            // Should only be started once.
+            Contract.ThrowIfTrue(_lazyConnection != null);
 
-            _workQueueSource.SetResult(
-                new AsyncBatchingWorkQueue<DocumentAndComments>(
-                TimeSpan.FromSeconds(1),
-                ProcessTodoCommentInfosAsync,
-                _asyncListener,
-                cancellationToken));
+            var cancellationToken = _disposalToken;
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Features/TodoComments/TodoCommentsListener.cs:76]: not sure why this was lazily created.

File: src/VisualStudio/Core/Def/TodoComments/VisualStudioTodoCommentsService.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/TodoComments/VisualStudioTodoCommentsService.cs:90]: Yes.  taht woudl make a ton of sense.  let me see what we can do about that.

**Pull request merged**